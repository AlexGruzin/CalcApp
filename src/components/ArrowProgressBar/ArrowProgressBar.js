import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';

import classNames from 'classnames';

import './ArrowProgressBar.styles.less';

const items = [
  {
    number: 0,
    label: 'Step 1',
    description: 'Connect to our Catalogue',
  },
  {
    number: 1,
    label: 'Step 2',
    description: 'Connect to our Analytics',
  },
  {
    number: 2,
    label: 'Step 3',
    description: 'Connect to our Channels',
  },
];

export const ArrowProgressBar = React.memo(({ activeIndex, className, handleSort }) => {
  const [actualItems, setActualItems] = useState([]);

  useEffect(() => {
    if (handleSort) setActualItems(items.sort(handleSort));
    const updatedItems = items.map((item, index) => ({
      ...item,
      isActive: index === activeIndex,
      isPast: index < activeIndex,
      isFuture: index > activeIndex,
    }));
    setActualItems(updatedItems);
  }, [activeIndex, handleSort]);

  return (
    <div className={classNames('arrow-bar', className)}>
      {actualItems.length &&
        actualItems.map((item, index) => {
          const { isActive, isPast, isFuture } = item;
          return (
            <div
              key={item.label}
              className={classNames('arrow-bar__item', { isActive, isPast, isFuture })}
              style={{ zIndex: items.length - index }}
            >
              <div className={classNames('arrow-bar__number')}>{item.number + 1}</div>
              <div className="arrow-bar__content">
                <div className="arrow-bar__label">{item.label}</div>
                <div className="arrow-bar__description">{item.description}</div>
              </div>
            </div>
          );
        })}
    </div>
  );
});

ArrowProgressBar.propTypes = {
  className: PropTypes.string,
  handleSort: PropTypes.func,
  activeIndex: PropTypes.number.isRequired,
};

ArrowProgressBar.defaultProps = {
  className: '',
  handleSort: null,
};

export default ArrowProgressBar;
