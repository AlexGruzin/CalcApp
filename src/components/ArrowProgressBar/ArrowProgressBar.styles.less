@import "~styles/generic.less";

.arrow-bar {
  display: flex;
  justify-content: space-between;
}

.arrow-bar__item {
  display: flex;
  position: relative;
  width: 33.33%;
  padding: var(--space-x3) var(--space-x4);
  border: 1px solid var(--black);
  border-right: none;
  border-left: none;
  cursor: default;

  &:before {
    content: '';
    display: block;
    position: absolute;
    width: 43px;
    height: 42px;
    top: 50%;
    right: -22px;
    transform: translateY(-50%) rotate(45deg) skew(10deg, 10deg);
    border-top: 1px solid var(--black);
    border-right: 1px solid var(--black);
    border-left: 1px solid transparent;
    border-bottom: 1px solid transparent;
    background: var(--white);
  }

  &:nth-child(1) {
    border-left: 1px solid var(--black);
  }

  &:nth-child(2),
  &:nth-child(3) {
    padding-left: var(--space-x10);
  }

  &:nth-child(3) {
    border-right: 1px solid var(--black);
  }

  &:nth-child(3):before {
    display: none;
  }
}

.arrow-bar__number {
  .montserrat-bold();
  padding-right: var(--space-x3);
  font-size: var(--fontSize-extraLarge);
}

.arrow-bar__content {
  display: flex;
  flex-direction: column;
}

.arrow-bar__label {
  margin-top: 0;
  margin-bottom: var(--space-x2);
}

.arrow-bar__description {
  margin: 0;
}

.arrow-bar__item.isActive {
  background: var(--purple);
  border-color: var(--purple);

  &:before {
    background: var(--purple);
    border-color: var(--purple);
  }

  .arrow-bar__number,
  .arrow-bar__label,
  .arrow-bar__description {
    color: var(--white);
  }
}
