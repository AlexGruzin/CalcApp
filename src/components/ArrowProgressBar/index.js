import ArrowProgressBar from './ArrowProgressBar';

export default ArrowProgressBar;
