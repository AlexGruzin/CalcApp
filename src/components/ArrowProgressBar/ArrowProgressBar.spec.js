import React from 'react';
import { render } from '@testing-library/react';

import ArrowProgressBar from './ArrowProgressBar';

describe('ArrowProgressBar tests', () => {
  it('Check snapshots of ArrowProgressBar are equal.', () => {
    const { asFragment } = render(<ArrowProgressBar items={[]} active={0} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
