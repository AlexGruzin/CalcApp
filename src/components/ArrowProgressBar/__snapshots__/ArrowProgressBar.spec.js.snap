// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowProgressBar tests Check snapshots of ArrowProgressBar are equal. 1`] = `
<DocumentFragment>
  <div
    class="arrow-bar"
  >
    <div
      class="arrow-bar__item"
      style="z-index: 3;"
    >
      <div
        class="arrow-bar__number"
      >
        1
      </div>
      <div
        class="arrow-bar__content"
      >
        <div
          class="arrow-bar__label"
        >
          Step 1
        </div>
        <div
          class="arrow-bar__description"
        >
          Connect to our Catalogue
        </div>
      </div>
    </div>
    <div
      class="arrow-bar__item"
      style="z-index: 2;"
    >
      <div
        class="arrow-bar__number"
      >
        2
      </div>
      <div
        class="arrow-bar__content"
      >
        <div
          class="arrow-bar__label"
        >
          Step 2
        </div>
        <div
          class="arrow-bar__description"
        >
          Connect to our Analytics
        </div>
      </div>
    </div>
    <div
      class="arrow-bar__item"
      style="z-index: 1;"
    >
      <div
        class="arrow-bar__number"
      >
        3
      </div>
      <div
        class="arrow-bar__content"
      >
        <div
          class="arrow-bar__label"
        >
          Step 3
        </div>
        <div
          class="arrow-bar__description"
        >
          Connect to our Channels
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
